tests/subsys/bluetooth/fast_pair/storage/factory_reset/src/main.c

```diff
@@ -0,0 +1,516 @@
+	zassert_ok(err, "Unexpected error in settings load");
+
+	cu_account_keys_validate_loaded(seed, key_count);
+	personalized_name_validate_loaded(name);
+}
+
+static void test_store(void)
+{
+	storage_validate();
+
+	mark_test_as_finished(RESET_TEST_FINISHED_STORE_BIT_POS);
+}
+
+static void store_data_and_factory_reset(void)
+{
+	static const uint8_t seed = 50;
+	static const uint8_t key_count = ACCOUNT_KEY_MAX_CNT;
+	static const char *name = "store_data_and_factory_reset";
+	int err;
+
+	cu_account_keys_generate_and_store(seed, key_count);
+	personalized_name_store(name);
+
+	err = bt_fast_pair_factory_reset();
+	zassert_ok(err, "Unexpected error in factory reset");
+}
+
+static void test_normal_reset(void)
+{
+	store_data_and_factory_reset();
+
+	storage_validate();
+
+	mark_test_as_finished(RESET_TEST_FINISHED_NORMAL_RESET_BIT_POS);
+}
+
+static void test_interrupted_reset_begin(void)
+{
+	storage_reset_pass = true;
+
+	store_data_and_factory_reset();
+
+	/* The recent instruction should trigger reboot so this code should never be entered. */
+	ztest_test_fail();
+}
+
+static void test_interrupted_reset_begin_teardown_fn(void) {}
+
+static void test_interrupted_reset_finish_setup_fn(void)
+{
+	int err;
+
+	err = settings_load();
+	zassert_ok(err, "Unexpected error in settings load");
+}
+
+static void test_interrupted_reset_finish(void)
+{
+	storage_validate();
+}
+
+static void normal_test_stage_finalize(void)
+{
+	int err;
+
+	for (size_t i = 0; i < RESET_TEST_FINISHED_COUNT; i++) {
+		if (!(test_finished_mask & BIT(i))) {
+			/* Do not reboot the device, clean up and abandon the test to
+			 * properly report fail in recent tests.
+			 */
+			err = all_settings_delete();
+			__ASSERT(!err, "Unable to delete settings (err %d)", err);
+
+			return;
+		}
+	}
+
+	test_stage = RESET_TEST_STAGE_INTERRUPTED;
+	err = settings_save_one(SETTINGS_RESET_TEST_STAGE_FULL_NAME, &test_stage,
+				sizeof(test_stage));
+	__ASSERT(!err, "Unexpected error in test settings (err %d)", err);
+
+	/* Reboot to finish the interrupted reset test. */
+	sys_reboot(SYS_REBOOT_WARM);
+}
+
+void test_main(void)
+{
+	int err;
+
+	err = settings_subsys_init();
+	__ASSERT(!err, "Error while initializing settings subsys (err %d)", err);
+
+	err = settings_load_subtree(SETTINGS_RESET_TEST_SUBTREE_NAME);
+	__ASSERT(!err, "Error while loading test settings (err %d)", err);
+
+	switch (test_stage) {
+	case RESET_TEST_STAGE_NORMAL:
+		/* The test_interrupted_reset_begin test must be placed last in the suite, it
+		 * triggers sys_reboot.
+		 */
+		ztest_test_suite(fast_pair_storage_factory_reset_normal_tests,
+				 ztest_unit_test(test_section_placement),
+				 ztest_unit_test_setup_teardown(test_store, setup_fn, teardown_fn),
+				 ztest_unit_test_setup_teardown(test_normal_reset, setup_fn,
+								teardown_fn),
+				 ztest_unit_test_setup_teardown(
+							test_interrupted_reset_begin,
+							setup_fn,
+							test_interrupted_reset_begin_teardown_fn)
```

Comment:
Would it work with the NULL pointer?
Have you checked if there unit test declaration with the setup part but without the teardown part e.g. ```ztest_unit_test_setup```?

I think you can use the ```unit_test_noop``` function here that is defined in the ztest.h header

Reply:
`unit_test_noop` is defined as static and not included in documentation, but it can be used and I see that many other tests use it that way so I will use it too.